pkgs/development/python-modules/pytest-asyncio/default.nix

```diff
@@ -9,16 +9,16 @@
 
 buildPythonPackage rec {
   pname = "pytest-asyncio";
-  version = "0.21.1"; # N.B.: when updating, tests bleak and aioesphomeapi tests
-  format = "pyproject";
+  version = "0.23.3"; # N.B.: when updating, tests bleak and aioesphomeapi tests
```

Comment:
Did you test these?

Reply:
Yes, pretty sure I did.

Reply:
aioesphomeapi test works
bleak tests fail hard because it wants to import a darwin-only library on linux.

```
bleak> ============================= test session starts ==============================
bleak> platform linux -- Python 3.12.2, pytest-7.4.4, pluggy-1.4.0
bleak> rootdir: /build/source
bleak> plugins: asyncio-0.23.3
bleak> asyncio: mode=Mode.STRICT
bleak> collected 0 items                                                              
bleak> INTERNALERROR> Traceback (most recent call last):
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/python.py", line 617, in _importtestmodule
bleak> INTERNALERROR>     mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
bleak> INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/pathlib.py", line 567, in import_path
bleak> INTERNALERROR>     importlib.import_module(module_name)
bleak> INTERNALERROR>   File "/nix/store/n3jf1lkdfakxskzsm4nlhss8mdgmcqhc-python3-3.12.2/lib/python3.12/importlib/__init__.py", line 90, in import_module
bleak> INTERNALERROR>     return _bootstrap._gcd_import(name[level:], package, level)
bleak> INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
bleak> INTERNALERROR>   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
bleak> INTERNALERROR>   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
bleak> INTERNALERROR>   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
bleak> INTERNALERROR>   File "<frozen importlib._bootstrap_external>", line 995, in exec_module
bleak> INTERNALERROR>   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
bleak> INTERNALERROR>   File "/build/source/bleak/backends/corebluetooth/__init__.py", line 9, in <module>
bleak> INTERNALERROR>     import objc
bleak> INTERNALERROR> ModuleNotFoundError: No module named 'objc'
bleak> INTERNALERROR> 
bleak> INTERNALERROR> The above exception was the direct cause of the following exception:
bleak> INTERNALERROR> 
bleak> INTERNALERROR> Traceback (most recent call last):
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/main.py", line 271, in wrap_session
bleak> INTERNALERROR>     session.exitstatus = doit(config, session) or 0
bleak> INTERNALERROR>                          ^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/main.py", line 324, in _main
bleak> INTERNALERROR>     config.hook.pytest_collection(session=session)
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
bleak> INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
bleak> INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
bleak> INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
bleak> INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_callers.py", line 181, in _multicall
bleak> INTERNALERROR>     return outcome.get_result()
bleak> INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_result.py", line 99, in get_result
bleak> INTERNALERROR>     raise exc.with_traceback(exc.__traceback__)
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
bleak> INTERNALERROR>     res = hook_impl.function(*args)
bleak> INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/main.py", line 335, in pytest_collection
bleak> INTERNALERROR>     session.perform_collect()
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/main.py", line 675, in perform_collect
bleak> INTERNALERROR>     self.items.extend(self.genitems(node))
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/main.py", line 842, in genitems
bleak> INTERNALERROR>     rep = collect_one_node(node)
bleak> INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/runner.py", line 546, in collect_one_node
bleak> INTERNALERROR>     ihook.pytest_collectstart(collector=collector)
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 501, in __call__
bleak> INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
bleak> INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_manager.py", line 119, in _hookexec
bleak> INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
bleak> INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_callers.py", line 138, in _multicall
bleak> INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
bleak> INTERNALERROR>   File "/nix/store/jdmzgfb4qvvb3k473ilyc34xp04g4yp7-python3.12-pluggy-1.4.0/lib/python3.12/site-packages/pluggy/_callers.py", line 102, in _multicall
bleak> INTERNALERROR>     res = hook_impl.function(*args)
bleak> INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/1zgxb751hj792d2ll4fkmw5rljgib77h-python3.12-pytest-asyncio-0.23.3/lib/python3.12/site-packages/pytest_asyncio/plugin.py", line 626, in pytest_collectstart
bleak> INTERNALERROR>     pyobject = collector.obj
bleak> INTERNALERROR>                ^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/python.py", line 310, in obj
bleak> INTERNALERROR>     self._obj = obj = self._getobj()
bleak> INTERNALERROR>                       ^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/python.py", line 528, in _getobj
bleak> INTERNALERROR>     return self._importtestmodule()
bleak> INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^
bleak> INTERNALERROR>   File "/nix/store/srq6cpawjsy99hk77ws2hl1fxlzvydx3-python3.12-pytest-7.4.4/lib/python3.12/site-packages/_pytest/python.py", line 642, in _importtestmodule
bleak> INTERNALERROR>     raise self.CollectError(
bleak> INTERNALERROR> _pytest.nodes.Collector.CollectError: ImportError while importing test module '/build/source/bleak/backends/corebluetooth/__init__.py'.
bleak> INTERNALERROR> Hint: make sure your test modules/packages have valid Python names.
bleak> INTERNALERROR> Traceback:
bleak> INTERNALERROR> /nix/store/n3jf1lkdfakxskzsm4nlhss8mdgmcqhc-python3-3.12.2/lib/python3.12/importlib/__init__.py:90: in import_module
bleak> INTERNALERROR>     return _bootstrap._gcd_import(name[level:], package, level)
bleak> INTERNALERROR> bleak/backends/corebluetooth/__init__.py:9: in <module>
bleak> INTERNALERROR>     import objc
bleak> INTERNALERROR> E   ModuleNotFoundError: No module named 'objc'
bleak> 
```

Reply:
Ok, we need to update to 0.23.5.

Reply:
Sorry, I guess I misremembered.